plugin.video.montreal.greek-tv/resources/lib/youtube.py

```diff
@@ -0,0 +1,352 @@
+# -*- coding: utf-8 -*-
+
+'''
+    Montreal Greek TV Add-on
+    Author: greektimes
+
+        License summary below, for more details please read license.txt file
+
+        This program is free software: you can redistribute it and/or modify
+        it under the terms of the GNU General Public License as published by
+        the Free Software Foundation, either version 2 of the License, or
+        (at your option) any later version.
+        This program is distributed in the hope that it will be useful,
+        but WITHOUT ANY WARRANTY; without even the implied warranty of
+        MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+        GNU General Public License for more details.
+        You should have received a copy of the GNU General Public License
+        along with this program.  If not, see <http://www.gnu.org/licenses/>.
+'''
+from __future__ import absolute_import, division, unicode_literals
+
+import re, json
+from resources.lib.compat import urlparse, parse_qs, quote_plus, range
+from resources.lib import workers, control, directory
+import requests
+from CommonFunctions import parseDOM
+
+
+def yt_resolve(setting='yt_resolve'):
+
+    resolve_bool = control.setting(setting) == 'true' and any(
+        [
+            control.condVisibility('System.HasAddon(script.module.{0})'.format('urlresolver')),
+            control.condVisibility('System.HasAddon(script.module.{0})'.format('resolveurl'))
+        ]
+    )
+
+    return resolve_bool
+
+
+class youtube(object):
+
+    def __init__(self, key='', api_key_setting='yt_api_key', yt_resolve_setting='yt_resolve'):
+
+        self.list = [];  self.data = []
+
+        self.base_link = 'http://www.youtube.com/'
+        self.base_addon = 'plugin://plugin.video.youtube/'
+        self.google_base_link = 'https://www.googleapis.com/youtube/v3/'
+
+        self.key_link = '&key={0}'.format(control.setting(api_key_setting) or key)
+
+        self.playlists_link = self.google_base_link + 'playlists?part=snippet&maxResults=50&channelId=%s'
+        self.playlist_link = self.google_base_link + 'playlistItems?part=snippet&maxResults=50&playlistId=%s'
+        self.videos_link = self.google_base_link + 'search?part=snippet&order=date&maxResults=50&channelId=%s'
+        self.content_link = self.google_base_link + 'videos?part=contentDetails&id=%s'
+        self.search_link = self.google_base_link + 'search?part=snippet&type=video&maxResults=5&q=%s'
+        self.youtube_search = self.google_base_link + 'search?q='
+
+        if yt_resolve(yt_resolve_setting):
```

Comment:
plugin playable urls are static in youtube addon, you don't need to resolve them. Also the youtube addon provides a builtin and importable script module you can use to resolve urls (e.g: https://github.com/enen92/script.teststream). As an alternative you can use script.module.youtube_dl

Reply:
will remove as above.